app/models/spree/order.rb

```diff
@@ -517,6 +465,21 @@ def create_proposed_shipments
       shipments
     end
 
+    # Clear shipments and move order back to address state
+    def ensure_updated_shipments
+      if !self.completed? && shipments.any?
+        shipments.destroy_all
+        restart_checkout_flow
+      end
+    end
+
+    def restart_checkout_flow
+      self.update_columns(
+        state: checkout_steps.first,
+        updated_at: Time.zone.now,
+      )
+    end
```

Comment:
hmm, I think this is https://refactoring.guru/smells/feature-envy this restart_checkout_flow should go to service OrderCheckoutRestart or the module Order::Checkout itself. order.rb shouldnt have order state logic I think.

Reply:
This bit was pulled straight from Spree 2.2, but yes it probably wants to be somewhere else :ok_hand:

Reply:
Moved :+1: